File: src/Avalonia.Samples/CompleteApps/Avalonia.MusicStore/Views/MusicStoreWindow.axaml.cs
using System;
using Avalonia.Controls;
using Avalonia.MusicStore.Messages;
using Avalonia.Controls.Notifications;
using CommunityToolkit.Mvvm.Messaging;

namespace Avalonia.MusicStore.Views
{
    public partial class MusicStoreWindow : Window
    {
        public MusicStoreWindow()
        {
            InitializeComponent();

            WeakReferenceMessenger.Default.Register<MusicStoreWindow, MusicStoreClosedMessage>(this,
                static (w, m) => w.Close(m.SelectedAlbum));
            
            WeakReferenceMessenger.Default.Register<MusicStoreWindow, NotificationMessage>(this, static (w, m) =>
            {
                w.NotificationManager.CloseAll();
                w.NotificationManager.Show(m.Message, NotificationType.Warning, TimeSpan.FromSeconds(3));
            });
        }
    }
}

File: src/Avalonia.Samples/FSharp/CompleteApps/MusicStore/Views/MusicStoreWindow.axaml.fs
namespace MusicStore.Views

open Avalonia
open Avalonia.Controls
open Avalonia.Markup.Xaml
open CommunityToolkit.Mvvm.Messaging
open MusicStore.Messages

type MusicStoreWindow() as this =
    inherit Window ()
    
    let onMusicStoreClosed (w: MusicStoreWindow) (msg: MusicStoreClosedMessage) =
        // Handle the music store closed message, e.g., retrieve the selected album
        let selectedAlbum = msg.SelectedAlbum
        // Perform actions with the selected album as needed
        // For example, you might want to display its details or add it to a collection
        w.Close selectedAlbum

    do 
        this.InitializeComponent()
        // Ensure the window and its child views have a proper DataContext at runtime
        // this.DataContext <- MusicStoreViewModel()
        WeakReferenceMessenger.Default.Register(this, onMusicStoreClosed)

    member private this.InitializeComponent() =
#if DEBUG
        this.AttachDevTools()
#endif
        AvaloniaXamlLoader.Load(this)
